radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,221 @@
+
+
+/**
+ * Constant Values for I2C Communication
+ */
+/** I2C Slave Address for EPS */
+#define BATTERY_I2C_SLAVE_ADDR 			(0x2A)
+/** Data Sent to the EPS via I2C for telemetry commands is 6 Bytes, 2 for the command, 4 for data*/
+#define BATTERY_TELEM_COMMAND_LENGTH	(6)
+/** Data sent to the EPS via I2C for all other commands is 4 Bytes, 2 for the command, 2 for data*/
+#define BATTERY_COMMAND_LENGTH			(4)
+/** Data Returned from the EPS via I2C is 4 Bytes */
+#define BATTERY_RESPONSE_LENGTH			(4)
+/** Delay between telemetry read/write operations with the EPS on I2C is 5ms */
+#define BATTERY_I2C_TELEM_DELAY			(5)
+/** Delay between other read/write operations with the EPS on I2C is 1ms */
+#define BATTERY_I2C_DELAY				(1)
+
+
+/***************************************************************************************************
+                                          PRIVATE GLOBALS
+***************************************************************************************************/
+
+/**
+ * 3 Byte commands for requesting output voltage readings for each bus
+ */
+static uint32_t batteryVoltageCommandBytes[3] = {
+		0x10E280,		// Output Voltage of Battery
+		0x10E210,		// Output Voltage of 5V Bus
+		0x10E200		// Output Voltage of 3.3V Bus
+};
+
+/**
+ * 3 Byte commands for requesting output current readings for each bus
+ */
+static uint32_t batteryCurrentCommandBytes[3] = {
+		0x10E284,		// Output Current of Battery Bus in mA
+		0x10E214,		// Output Current of 5V Bus in mA
+		0x10E204		// Output Current of 3.3V Bus in mA
+};
+
+/**
+ * 4 Byte commands for requesting temperature readings for each board
+ */
+static uint32_t batteryTemperatureCommandBytes[4] = {
+		0x10E308,		// Motherboard Temperature
+		0x10E398,		// Daughterboard 1 Temperature
+		0x10E3A8,		// Daughterboard 2 Temperature
+		0x10E3B8		// Daughterboard 3 Temperature
+};
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static uint16_t batteryI2cTalk(uint32_t command);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Request and store all of the relavant data (voltage, current, temperature) from the battery board
+ * 		and battery daughter boards
+ * @return A BatteryStatus object containing all of the telemetry of the battery board
+ */
+BatteryStatus getBatteryTelemetry(void){
+
+	// Create a new BatteryStatus Structure
+	BatteryStatus dataStorage;
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_received;
+
+
+	// Get ADC Output Voltages
+	i2c_received = batteryI2cTalk(batteryVoltageCommandBytes[0]);
```

Comment:
implement an error check or else you are loading your struct with error codes instead of data